common/src/main/scala/models/ShardedNotification.scala

```diff
@@ -13,6 +15,7 @@ object ShardRange {
 case class ShardedNotification(
   notification: Notification,
   range: ShardRange,
+  notificationAppReceivedTime: Option[Instant]
```

Comment:
We are touching a lot of files here in order to wire this timestamp throughout the codebase. It might be easiest to review all of these via this commit: https://github.com/guardian/mobile-n10n/commit/a8d26d171c249e229a2240bf472d9d183866d3c2.